src/ZuluSCSI_disk.cpp

```diff
@@ -398,7 +425,7 @@ bool scsiDiskOpenHDDImage(int target_idx, const char *filename, int scsi_id, int
         PLATFORM_CONFIG_HOOK(&img);
 #endif
 
-        setDefaultDriveInfo(target_idx);
```

Comment:
I think the filename-based name setting could be a separate function instead of putting it into `setDefaultDriveInfo()`. If you call it first, `setDefaultDriveInfo()` will detect that the vendor/product name is already set.

Something like this:

1. Add `bool name_from_image` to `image_config_t`.
2. In `scsiDiskLoadConfig()` use `ini_getbool()` to get the setting from ini file
3. In here, before `setDefaultDriveInfo()` is called, do `if (img.name_from_image) { setNameFromImage(img, filename); }`